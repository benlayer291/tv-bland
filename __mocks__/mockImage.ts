import { InterfaceTvShowImage } from '../types/interfaces'

export const mockImage: InterfaceTvShowImage = {
  medium:
    'http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg',
  original:
    'http://static.tvmaze.com/uploads/images/original_untouched/81/202627.jpg',
}
