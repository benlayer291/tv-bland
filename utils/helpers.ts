export const removeProtocolFromUrl = (url: string): string =>
  url.replace(/^https?:\/\//i, '//')

export const removeHtmlTagsFromString = (htmlString: string): string =>
  htmlString.replace(/(<([^>]+)>)/gi, '')

export const checkDeviceIsMobile = (): boolean => {
  return (
    typeof navigator !== 'undefined' &&
    /Android|Mobi/i.test(navigator.userAgent)
  )
}
