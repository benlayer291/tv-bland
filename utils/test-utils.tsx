import { render, RenderOptions } from '@testing-library/react'

type Props = {
  children: React.ReactElement
}

const Providers: React.FunctionComponent<Props> = ({ children }: Props) => {
  return <>{children}</>
}

const customRender = (
  ui: React.ReactElement,
  options: Omit<RenderOptions, 'queries'> = {}
) => render(ui, { wrapper: Providers, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
