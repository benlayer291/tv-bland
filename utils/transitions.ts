type Transtion = {
  duration: number
  ease: number[]
}

export const transition: Transtion = {
  duration: 5,
  ease: [0.43, 0.13, 0.23, 0.96],
}
