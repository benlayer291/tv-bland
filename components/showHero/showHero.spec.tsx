import React from 'react'
import { render, RenderResult } from '../../utils/test-utils'
import userEvent from '@testing-library/user-event'

import { mockShow } from '../../__mocks__/mockShow'

import ShowHero from './showHero'

describe('<ShowHero />', () => {
  let el: RenderResult

  it('should render', () => {
    el = render(<ShowHero show={mockShow} />)

    expect(el.container).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    el = render(<ShowHero show={mockShow} />)

    expect(el.container).toMatchSnapshot()
  })

  describe('given show has long decription', () => {
    it('should render with "Read More" button', () => {
      el = render(<ShowHero show={mockShow} />)

      expect(el.queryByText(/Read More/)).toBeInTheDocument()
    })
  })

  describe('given show has "Read More" button', () => {
    it('should toggle "Read More" to "Read Less" ', async () => {
      el = render(<ShowHero show={mockShow} />)

      await userEvent.click(el.getByRole('button'))

      expect(el.queryByText(/Read Less/)).toBeInTheDocument()
    })
  })
})
