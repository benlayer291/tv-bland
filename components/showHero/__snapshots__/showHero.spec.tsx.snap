// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowHero /> should match snapshot 1`] = `
<div>
  <div
    class="ShowHero"
  >
    <div
      class="wrapper"
    >
      <div
        class="ShowHero__container"
      >
        <div
          class="ShowHero__imageContainer"
        >
          <div
            class="ShowHero__imageContainerInner"
          >
            <div
              class="ShowImage ShowHero__image"
              data-testid="show-image"
            >
              <div
                class="BaseRevealImage"
                data-testid="base-reveal-image"
              >
                <img
                  alt="Cover image for Under the Dome"
                  class="BaseRevealImage__image"
                  height="295"
                  loading="eager"
                  src="//static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg"
                  width="210"
                />
                <img
                  alt="Cover image for Under the Dome"
                  class="BaseRevealImage__filter"
                  height="295"
                  loading="eager"
                  src="//static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg"
                  width="210"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="ShowHero__infoContainer"
        >
          <div
            class="StarRating ShowHero__rating StarRating--is-large"
            data-testid="rating"
          >
            <div
              class="StarRating__icons"
            >
              <svg
                class="Star StarRating__icon Star--is-active"
                data-testid="star"
                viewBox="0 0 14 12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M6.667 9.95L2.547 12l.786-4.342L0 4.584l4.607-.634L6.667 0l2.06 3.95 4.606.634L10 7.658 10.787 12l-4.12-2.05z"
                  fill-rule="evenodd"
                />
              </svg>
              <svg
                class="Star StarRating__icon Star--is-active"
                data-testid="star"
                viewBox="0 0 14 12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M6.667 9.95L2.547 12l.786-4.342L0 4.584l4.607-.634L6.667 0l2.06 3.95 4.606.634L10 7.658 10.787 12l-4.12-2.05z"
                  fill-rule="evenodd"
                />
              </svg>
              <svg
                class="Star StarRating__icon Star--is-active"
                data-testid="star"
                viewBox="0 0 14 12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M6.667 9.95L2.547 12l.786-4.342L0 4.584l4.607-.634L6.667 0l2.06 3.95 4.606.634L10 7.658 10.787 12l-4.12-2.05z"
                  fill-rule="evenodd"
                />
              </svg>
              <svg
                class="Star StarRating__icon"
                data-testid="star"
                viewBox="0 0 14 12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M6.667 9.95L2.547 12l.786-4.342L0 4.584l4.607-.634L6.667 0l2.06 3.95 4.606.634L10 7.658 10.787 12l-4.12-2.05z"
                  fill-rule="evenodd"
                />
              </svg>
              <svg
                class="Star StarRating__icon"
                data-testid="star"
                viewBox="0 0 14 12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M6.667 9.95L2.547 12l.786-4.342L0 4.584l4.607-.634L6.667 0l2.06 3.95 4.606.634L10 7.658 10.787 12l-4.12-2.05z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <span
              class="StarRating__text"
              data-testid="rating-text"
            >
              3.25/5
            </span>
          </div>
          <h1>
            Under the Dome
          </h1>
          <div
            class="ShowHero__descriptionContainer"
            style="max-height: 60px;"
          >
            <div
              class="ShowHero__description"
            >
              <p>
                <b>
                  Under the Dome
                </b>
                 is the story of a small town that is suddenly and inexplicably sealed off from the rest of the world by an enormous transparent dome. The town's inhabitants must deal with surviving the post-apocalyptic conditions while searching for answers about the dome, where it came from and if and when it will go away.
              </p>
            </div>
          </div>
          <button
            class="BaseButton ShowHero__toggle"
            type="button"
          >
            Read More
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
